src/main/java/io/coala/jetbrains/settings/ProjectSettings.java

```diff
@@ -73,4 +88,41 @@ public void removeSectionFromFilter(String section) {
     public void clearSections() {
         this.sections.clear();
     }
+
+    @Nullable
+    public Path determineAndSetExecutable(String executable) throws ExecutionException, InterruptedException, IOException {
+        final GeneralCommandLine commandLine = new GeneralCommandLine().withParameters(executable);
+        if(SystemInfo.isUnix) {
+            commandLine.setExePath("/usr/bin/which");
+        } else if(SystemInfo.isWindows) {
+            commandLine.setExePath("C:\\Windows\\System32\\where.exe");
+        }
+        Process process = commandLine.createProcess();
+        process.waitFor();
+        if(process.exitValue() == 0) {
+            String parsed_executable = IOUtils.toString(process.getInputStream(), Charset.defaultCharset()).trim();
+            return Paths.get(parsed_executable);
+        }
+        return null;
+    }
+
+    private void runAutomaticExecutableRetrieval() {
+        final Application app = ApplicationManager.getApplication();
+        app.executeOnPooledThread(() -> {
+            app.invokeLater(() -> {
+                try {
+                    Path coalaExecutableLocation = determineAndSetExecutable("coala");
+                    if(coalaExecutableLocation != null) {
+                        executable = coalaExecutableLocation.toString();
+                    }
+                } catch (ExecutionException e) {
```

Comment:
I believe starting Java 7 you can have multiple typed catches in a single block.

Reply:
Now that error made sense, IntelliJ was showing `catch block same as previous` and I thought yes it is but the exception I'm catching is different!